folly/experimental/JemallocHugePageAllocator.cpp

```diff
@@ -21,14 +21,18 @@
 
 #include <sstream>
 
-#if defined(MADV_HUGEPAGE) && defined(FOLLY_USE_JEMALLOC) && !FOLLY_SANITIZE
+#if (defined(MADV_HUGEPAGE) || defined(MAP_ALIGNED_SUPER)) && defined(FOLLY_USE_JEMALLOC) && !FOLLY_SANITIZE
+#if !defined(__FreeBSD__)
 #include <jemalloc/jemalloc.h>
-#if (JEMALLOC_VERSION_MAJOR >= 5)
+#else
+#include <malloc_np.h>
+#endif
+#if defined(__FreeBSD__) || (JEMALLOC_VERSION_MAJOR >= 5)
```

Comment:
Why do we need the `__FreeBSD__` check here? Does jemalloc on FreeBSD not define `JEMALLOC_VERSION_MAJOR`?

Reply:
Nope. they keep all "close to the vest". fair enough for the rest.